spring-cloud-alibaba-sentinel-gateway/src/main/java/org/springframework/cloud/alibaba/sentinel/gateway/scg/SentinelSCGAutoConfiguration.java

```diff
@@ -0,0 +1,155 @@
+import org.springframework.web.reactive.function.server.ServerResponse;
+import org.springframework.web.reactive.result.view.ViewResolver;
+import org.springframework.web.server.ServerWebExchange;
+
+import com.alibaba.csp.sentinel.adapter.gateway.common.SentinelGatewayConstants;
+import com.alibaba.csp.sentinel.adapter.gateway.sc.SentinelGatewayFilter;
+import com.alibaba.csp.sentinel.adapter.gateway.sc.callback.BlockRequestHandler;
+import com.alibaba.csp.sentinel.adapter.gateway.sc.callback.GatewayCallbackManager;
+import com.alibaba.csp.sentinel.adapter.gateway.sc.callback.RedirectBlockRequestHandler;
+import com.alibaba.csp.sentinel.adapter.gateway.sc.exception.SentinelGatewayBlockExceptionHandler;
+import com.alibaba.csp.sentinel.config.SentinelConfig;
+import com.alibaba.csp.sentinel.util.StringUtil;
+
+import reactor.core.publisher.Mono;
+
+/**
+ * @author <a href="mailto:fangjian0423@gmail.com">Jim</a>
+ */
+@Configuration
+@ConditionalOnClass(GlobalFilter.class)
+@ConditionalOnProperty(prefix = ConfigConstants.GATEWAY_PREFIX, name = "enabled", havingValue = "true", matchIfMissing = true)
+@EnableConfigurationProperties(SentinelGatewayProperties.class)
+public class SentinelSCGAutoConfiguration {
+
+	private static final Logger logger = LoggerFactory
+			.getLogger(SentinelSCGAutoConfiguration.class);
+
+	private final List<ViewResolver> viewResolvers;
+	private final ServerCodecConfigurer serverCodecConfigurer;
+
+	@Autowired
+	private Optional<BlockRequestHandler> blockRequestHandlerOptional;
+
+	@Autowired
+	private SentinelGatewayProperties gatewayProperties;
+
+	@PostConstruct
+	private void init() {
+		// blockRequestHandlerOptional has low priority
+		blockRequestHandlerOptional.ifPresent(GatewayCallbackManager::setBlockHandler);
+		initAppType();
+		initFallback();
+	}
+
+	public SentinelSCGAutoConfiguration(
+			ObjectProvider<List<ViewResolver>> viewResolversProvider,
+			ServerCodecConfigurer serverCodecConfigurer) {
+		this.viewResolvers = viewResolversProvider.getIfAvailable(Collections::emptyList);
+		this.serverCodecConfigurer = serverCodecConfigurer;
+	}
+
+	private void initAppType() {
+		System.setProperty(SentinelConfig.APP_TYPE,
+				String.valueOf(SentinelGatewayConstants.APP_TYPE_GATEWAY));
+	}
+
+	private void initFallback() {
+		FallbackProperties fallbackProperties = gatewayProperties.getFallback();
+		if (fallbackProperties == null
+				|| StringUtil.isBlank(fallbackProperties.getMode())) {
+			return;
+		}
+		if (ConfigConstants.FALLBACK_MSG_RESPONSE.equals(fallbackProperties.getMode())) {
+			if (StringUtil.isNotBlank(fallbackProperties.getResponseBody())) {
+				GatewayCallbackManager.setBlockHandler(new BlockRequestHandler() {
+					@Override
+					public Mono<ServerResponse> handleRequest(ServerWebExchange exchange,
+							Throwable t) {
+						return ServerResponse
+								.status(fallbackProperties.getResponseStatus())
+								.contentType(MediaType.APPLICATION_JSON_UTF8)
```

Comment:
The response content type might not be JSON?

Reply:
yes, i think we can add a new configuration item `spring.cloud.sentinel.scg.fallback.content-type`, default value is `application/json;charset=UTF-8`.

How do you think about it?

Reply:
+1 for this